src/python/gudhi/periodic_cubical_complex.pyx

```diff
@@ -77,34 +100,49 @@ cdef class PeriodicCubicalComplex:
         """
 
     # The real cython constructor
-    def __cinit__(self, dimensions=None, top_dimensional_cells=None,
+    def __cinit__(self, dimensions=None, top_dimensional_cells=None, vertices=None,
                   periodic_dimensions=None, perseus_file=''):
         if ((dimensions is not None) and (top_dimensional_cells is not None)
-            and (periodic_dimensions is not None) and (perseus_file == '')):
-            self._construct_from_cells(dimensions, top_dimensional_cells, periodic_dimensions)
+            and (vertices is None) and (periodic_dimensions is not None)
+            and (perseus_file == '')):
+            self._construct_from_cells(dimensions, top_dimensional_cells, periodic_dimensions, True)
+        elif ((dimensions is not None) and (vertices is not None)
+            and (top_dimensional_cells is None) and (periodic_dimensions is not None)
+            and (perseus_file == '')):
+            self._construct_from_cells(dimensions, vertices, periodic_dimensions, False)
         elif ((dimensions is None) and (top_dimensional_cells is not None)
-            and (periodic_dimensions is not None) and (perseus_file == '')):
+            and (vertices is None) and (periodic_dimensions is not None)
+            and (perseus_file == '')):
             top_dimensional_cells = np.array(top_dimensional_cells,
                                              copy = False,
                                              order = 'F')
             dimensions = top_dimensional_cells.shape
             top_dimensional_cells = top_dimensional_cells.ravel(order='F')
-            self._construct_from_cells(dimensions, top_dimensional_cells, periodic_dimensions)
+            self._construct_from_cells(dimensions, top_dimensional_cells, periodic_dimensions, True)
+        elif ((dimensions is None) and (vertices is not None)
+            and (top_dimensional_cells is None) and (periodic_dimensions is not None)
+            and (perseus_file == '')):
+            vertices = np.array(vertices,
+                                copy = False,
+                                order = 'F')
+            dimensions = vertices.shape
+            vertices = vertices.ravel(order='F')
+            self._construct_from_cells(dimensions, vertices, periodic_dimensions, False)
```

Comment:
(comment in a random place)
```python
PeriodicCubicalComplex(vertices=[[2, 1, 6, 8],
                                 [5, 9, 4, 0],
                                 [4, 7, 3, 9]], periodic_dimensions=(True, True)).persistence()
```
outputs
```
[(2, (9.0, inf)),
 (1, (5.0, inf)),
 (1, (7.0, inf)),
 (1, (6.0, 8.0)),
 (0, (1.0, inf)),
 (0, (2.0, 4.0)),
 (0, (4.0, 5.0))]
```
I didn't check the whole diagram, but at least the infinite connected component should start at 0, so I have some debugging to do...

Reply:
Ah, it looks like Vertices_iterator, inherited from the non-periodic case, doesn't work with periodicity. In particular, I would expect `this->counter[dim] == this->b.sizes[dim]` to have a `- 1` in periodic dimensions.